Scripts/Web_Scrappers/Shopify_Web_Scraper/shopify.py

```diff
@@ -0,0 +1,234 @@
+
+USER_AGENT = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3)' \
+             ' AppleWebKit/537.36 (KHTML, like Gecko)' \
+             ' Chrome/35.0.1916.47 Safari/537.36'
+
+
+def get_page(url, page, collection_handle=None):
+    """
+    Get page contents
+    """
+    full_url = url
+    if collection_handle:
+        full_url += '/collections/{}'.format(collection_handle)
+    full_url += '/products.json'
+    req = urllib.request.Request(
+        full_url + '?page={}'.format(page),
+        data=None,
+        headers={
+            'User-Agent': USER_AGENT
+        }
+    )
+    while True:
+        try:
+            data = urllib.request.urlopen(req).read()
+            break
+        except HTTPError:
+            print('Blocked! Sleeping...')
+            time.sleep(180)
+            print('Retrying')
+
+    products = json.loads(data.decode())['products']
+    return products
+
+
+def get_page_collections(url):
+    """
+    Get Collections on the page using the json parsing
+    """
+    full_url = url + '/collections.json'
+    page = 1
+    while True:
+        req = urllib.request.Request(
+            full_url + '?page={}'.format(page),
+            data=None,
+            headers={
+                'User-Agent': USER_AGENT
+            }
+        )
+        while True:
+            try:
+                data = urllib.request.urlopen(req).read()
+                break
+            except HTTPError:
+                print('Blocked! Sleeping...')
+                time.sleep(180)
+                print('Retrying')
+
+        cols = json.loads(data.decode())['collections']
+        if not cols:
+            break
+        for collection in cols:
+            yield collection
+        page += 1
+
+
+def check_shopify(url):
+    try:
+        get_page(url, 1)
+        return True
+    except Exception:
+        return False
+
+
+def fix_url(url):
+    fixed_url = url.strip()
+    if not fixed_url.startswith('http://') and \
+            not fixed_url.startswith('https://'):
+        fixed_url = 'https://' + fixed_url
+
+    return fixed_url.rstrip('/')
+
+
+def extract_products_collection(url, col):
+    page = 1
+    products = get_page(url, page, col)
+    while products:
+        for product in products:
+            title = product['title']
+            print("Processing:", title)
+            product_type = product['product_type']
+            product_url = url + '/products/' + product['handle']
+            product_handle = product['handle']
+
+            def get_image(variant_id):
+                images = product['images']
+                for i in images:
+                    k = [str(v) for v in i['variant_ids']]
+                    if str(variant_id) in k:
+                        return i['src']
+
+                return ''
+
+            for i, variant in enumerate(product['variants']):
+                price = variant['price']
+                option1_value = variant['option1'] or ''
```

Comment:
simply use get method it will check for given key and return value otherwise return an predefined value
```suggestion
                option1_value = variant.get('option1','')
```